marlowe-cli/src/Language/Marlowe/CLI/Command/Parse.hs

```diff
@@ -295,22 +296,22 @@ parseParty =
     <|> O.readerError "Invalid party."
 
 
--- | Reader for `Party` `PK`.
+-- | Reader for `Party` `Address`.
 readPartyPkEither :: String               -- ^ The string to be read.
                   -> Either String Party  -- ^ Either the public key hash role or an error message.
 readPartyPkEither s =
-  case s =~ "^PK=([[:xdigit:]]{56})$" of
-    [[_, pubKeyHash]] -> PK <$> readPubKeyHashEither pubKeyHash
-    _                 -> Left "Invalid public key hash for party."
+  case deserialiseAddressBech32 $ T.pack s of
+    Just (network, address) -> Right $ Address network address
+    _                       -> Left "Invalid public key hash for party."
```

Comment:
Invalid bech32 address

Reply:
Thanks. I grepped through the code to look for other stray references to public key hashes.